lib/Browser.js

```diff
@@ -51,24 +59,53 @@ class Browser extends EventEmitter {
     return this._process;
   }
 
+  /**
+   * @return {!Promise<!BrowserContext>}
+   */
+  async createIncognitoBrowserContext() {
+    const {browserContextId} = await this._connection.send('Target.createBrowserContext');
+    const context = new BrowserContext(this, browserContextId);
+    this._contexts.set(browserContextId, context);
+    return context;
+  }
+
+  /**
+   * @return {!Array<!BrowserContext>}
+   */
+  browserContexts() {
+    return [this._defaultContext, ...Array.from(this._contexts.values())];
+  }
+
+  /**
+   * @param {?string} contextId
+   */
+  async _disposeContext(contextId) {
+    await this._connection.send('Target.disposeBrowserContext', {browserContextId: contextId || undefined});
+    this._contexts.delete(contextId);
+  }
+
   /**
    * @param {!Puppeteer.Connection} connection
+   * @param {!Array<string>} contextIds
    * @param {!BrowserOptions=} options
    * @param {?Puppeteer.ChildProcess} process
    * @param {function()=} closeCallback
    */
-  static async create(connection, options, process, closeCallback) {
-    const browser = new Browser(connection, options, process, closeCallback);
+  static async create(connection, contextIds, options, process, closeCallback) {
+    const browser = new Browser(connection, contextIds, options, process, closeCallback);
     await connection.send('Target.setDiscoverTargets', {discover: true});
     return browser;
   }
 
   /**
-   * @param {{targetInfo: !Puppeteer.TargetInfo}} event
+   * @param {!Protocol.Target.targetCreatedPayload} event
    */
   async _targetCreated(event) {
     const targetInfo = event.targetInfo;
-    const target = new Target(targetInfo, this, () => this._connection.createSession(targetInfo.targetId), this._ignoreHTTPSErrors, !this._appMode, this._screenshotTaskQueue);
+    const {browserContextId} = targetInfo;
+    const context = browserContextId && this._contexts.has(browserContextId) ? this._contexts.get(browserContextId) : this._defaultContext;
```

Comment:
can you put paren around `browserContextId && this._contexts.has(browserContextId)`. I never remember order of operations ternary vs boolean operator.

Reply:
Done. 

> I never remember order of operations ternary vs boolean operator

Interestingly I had the same issue, and eventually found that dealing with these kind of expressions in codebases help me memorize the rules. For example, there's a common pattern to handle nullable collections:

```js
const result = arr && arr.includes(value) ? 'yeah!' : 'nope.';
```